### PR TITLE
Proof of concept for automatic "can you hear me" response to CQ.

### PR DESCRIPTION
**Pull request by kronenpj**
Working proof of concept to satisfy Feature #156 request.

NOTE: This will behave "badly" on the air with prior releases. This release will respond to each of the 3 CQ packets prior release software sends.

### CODE REVIEW

**Review by DJ2LS — APPROVED** [review]

File: tnc/data_handler.py [hunk 1]
```diff
@@ -1665,6 +1670,68 @@ def received_cq(self, data_in:bytes):
...
+
+        Returns:
+            Nothing
+        """
+        # Sleep a random amount of time before responding to make it more likely to be
+        # heard when many stations respond. Each DATAC0 frame is 0.44 sec (440ms) in
+        # duration, plus overhead. Set the wait interval to be random between 0 and 2s
+        # in 0.5s increments. If the maximum time is "too long,"" the user may believe the
+        # UI / TNC has frozen because this sleep blocks all other activity.
+        time.sleep(randrange(0, 20, 5) / 10.0)
```
[the remark this answers is not in the document]
Comment (kronenpj) [reply tnc/data_handler.py]: Ah, didn't notice that one. Updated & pushed.

File: tnc/data_handler.py [hunk 2]
```diff
@@ -1665,6 +1670,68 @@ def received_cq(self, data_in:bytes):
...
+          data_in:bytes:
+
+        Returns:
+
+        """
+        # here we add the received station to the heard stations buffer
+        dxcallsign = helpers.bytes_to_callsign(bytes(data_in[1:9]))
+        dxgrid = bytes(data_in[9:13]).rstrip(b'\x00')
+
+        jsondata = {"type" : "qrv", "status" : "received",  "uuid" : str(uuid.uuid4()),  "timestamp": int(time.time()), "mycallsign" : str(self.mycallsign, 'utf-8'), "dxcallsign": str(dxcallsign, 'utf-8'), "dxgrid": str(dxgrid, 'utf-8'), "snr": str(static.SNR)}
```
[the remark this answers is not in the document]
Comment (kronenpj) [reply tnc/data_handler.py:1725]: That'll be interesting, but as you say, probably something to do later.

File: tnc/data_handler.py [hunk 3]
```diff
@@ -1665,6 +1670,67 @@ def received_cq(self, data_in:bytes):
...
+          data_in:bytes:
+
+        Returns:
+            Nothing
+        """
+        # Sleep a random amount of time before responding to make it more likely to be
+        # heard when many stations respond. Each DATAC0 frame is 0.44 sec (440ms) in
+        # duration, plus overhead. Set the wait interval to be random between 0 and 2s
+        # in 0.5s increments.
+        helpers.wait(randrange(0, 20, 5) / 10.0)
```
[the remark this answers is not in the document]
Comment (DJ2LS) [reply tnc/data_handler.py:1690]: I did a small tests - seems to work for me

**Review by DJ2LS — APPROVED** [review]

**Comment by DJ2LS** [comment]
@kronenpj Can I merge this so we can test this over the air? My local test here seems to be successful

**Comment by kronenpj** [comment]
I believe so. My tests were successful as well. Let's see what comments we receive from the alpha tester community :).

**Pull request merged**